DrawerKit/DrawerKit/Public API/Notifications.swift

```diff
@@ -12,13 +12,38 @@ public class NotificationToken {
     deinit { center.removeObserver(token) }
 }
 
+@available(*, deprecated, message: "Do not use this.")
 public protocol NotificationEnum {
     var name: Notification.Name { get }
 }
 
 extension NotificationCenter {
     private static let notificationKey = "notificationKey"
 
+    public func addObserver<T: DrawerNotificationInfo>(_ type: T.Type,
+                                                       object: Any? = nil,
+                                                       queue: OperationQueue? = nil,
+                                                       using block: @escaping (T) -> ()) -> NotificationToken {
+        let token = addObserver(forName: T.name, object: object, queue: queue, using: { note in
+            guard let note = note.userInfo?[NotificationCenter.notificationKey] as? T else {
+                fatalError("incorrect notification key type")
+            }
+            block(note)
+        })
+
+        return NotificationToken(token: token, center: self)
+    }
+
+    internal func post<T: DrawerNotificationInfo>(_ info: T, object: Any? = nil) {
+        let userInfo = [NotificationCenter.notificationKey: info]
+        post(name: T.name, object: object, userInfo: userInfo)
+    }
+}
+
+// MARK: - Deprecated
+
+extension NotificationCenter {
+    @available(*, deprecated, message: "Do not use this.")
```

Comment:
Why not `unavailable` if you want users not to use it?

Reply:
It's for having a migration period for end users, and also making a non-breaking changes.